src/App.tsx

```diff
@@ -96,14 +279,29 @@ export const Main = () => {
             <button
               className="bg-white text-dark-body text-sm px-4 py-2 border-light-body border-2 border-solid rounded-xl"
               onClick={() => {
-                console.log("Delete");
+
+                //Function for post deletion it see if there is any selected items else it does nothing
+                const selectedPost = posts.find((post) => post.isSelected);
+                if (selectedPost) {
+                  deletePost({ variables: { id: selectedPost.id } }).then(result => {
+                    if (result.data.deletePost) {
+                      const postsArray = [...posts];
+                      const idx = postsArray.indexOf(selectedPost);
+                      postsArray.splice(idx, 1);
+                      setPosts(postsArray);
```

Comment:
While this works, it create a really strange UX for the user. The interface seems _frozen_ and it takes some seconds when deleting the item.
How would you go about giving a smoother experience to the user?